### ISSUE TITLE
terraform locals not expanded for non-default provider

### ISSUE THREAD

**Issue by tommywo** [bridgecrewio/checkov#5118]
**Describe the issue**
We created a custom check based on `BaseProviderCheck` that checks the contents of `default_tags` block. 
It fails for every non-default provider, due to locals not being expanded. 

**Examples**
```hcl
locals {
  tags = {
    test         = "Test"
  }
}
provider "aws" {
  default_tags {
    tags = local.tags
  }
}


provider "aws" {
  alias = "test"
  default_tags {
    tags = local.tags
  }
}
```
conf is reported as: 
```
{'__end_line__': 6, '__start_line__': 1, 'default_tags': [{'tags': [{'test': 'Test'}]}], '__address__': 'aws.default'}
{'__end_line__': 19, '__start_line__': 13, 'alias': ['test'], 'default_tags': [{'tags': ['${local.tags}']}], '__address__': 'aws.test'}
```
**Desktop (please complete the following information):**
 - OS: mac os / linux
 - Checkov Version: 2.3.251

**Additional context**

**Comment by gruebel** [comment]
hey @tommywo thanks for reaching out.

I can verify it is a bug on our side.
